### DISCUSSION TITLE
Text area border?

### DISCUSSION THREAD

**Discussion by jaka1-newbie** [ImageMagick/ImageMagick#2212] [Help]
[Anne of Green Gables by L. M. Montgomery.pdf](https://github.com/ImageMagick/ImageMagick/files/4847645/Anne.of.Green.Gables.by.L.M.Montgomery.pdf)

![cover_pages240_6x9_white](https://user-images.githubusercontent.com/67605008/86047404-2722db80-ba4f-11ea-9718-81e3d684961a.png)
Want to create Cover Template  - Anne of Green Gables by L. M. Montgomery.pdf as example
I have an question - how place text on page (size of page we have) in right cover in pink borders(file cover_pages240 as example of cover tempplate). If title is short no problem, but Title can be long and how put border for text - to imagemagick draw it in 2 lines?

**Comment by fmw42** [comment]
Please show where you need to put the text title.  

Look at caption: at https://imagemagick.org/Usage/text/#caption Define the size of the box for the text and create a new image with transparent background of that size. Then composite the resulting image over your background image at the location you want.

**Comment by jaka1-newbie** [comment]
That's what I was looking for.
It remains to understand how to get into the middle of the right half.

As in pdf i need put Title in right part of cover. In image it is middle of front cover.

**Comment by jaka1-newbie** [comment]
One more question please. As an Great Man helping me.
Can we fill color 30% of image from up?

**Comment by fmw42** [comment]
I am sorry. But I still do not know where you want to put the title. Can you use GIMP or Photoshop or some other tool and put an example title where you want it. Also color the image as you want for 30%. I do not know what you want for that. Then put the image here so I can see what you are trying to do with ImageMagick

**Comment by jaka1-newbie** [comment]
As result I want to get like this
![Anne of Green Gables by L M Montgomery_1](https://user-images.githubusercontent.com/67605008/86053781-55a5b400-ba59-11ea-97b7-2fe18549a0cd.jpg)

**Comment by fmw42** [comment]
Sorry, I do not understand. How do you get that green image from the pink one your showed?  Which is the title you want? There are 3 sections of text. Which one is the one you want added.  Where is the color you want placed?  Please show the input before adding the text and before adding the color. Then show the desired output.

**Comment by jaka1-newbie** [comment]
How do you get that green image from the pink one your showed?
I did it manually. I found your programm. Trying to do this with it. Pink image invisible layer - pink border forbidden place for text - used for ease of placement. 

 Which is the title you want? There are 3 sections of text. Which one is the one you want added.
Main Title ANNE OF GREEN GABLES (anotheк will be similar.

 Where is the color you want placed? 
Please show the input before adding the text and before adding the color. Then show the desired output.

![1step](https://user-images.githubusercontent.com/67605008/86055783-9d7a0a80-ba5c-11ea-9227-de20af6f05b7.jpg)
![2step](https://user-images.githubusercontent.com/67605008/86055786-9e12a100-ba5c-11ea-8396-eaa68f2915b5.jpg)
![3 step](https://user-images.githubusercontent.com/67605008/86055787-9eab3780-ba5c-11ea-9e50-2fa322104ec6.jpg)
![4step](https://user-images.githubusercontent.com/67605008/86055789-9f43ce00-ba5c-11ea-8928-8d7668b7333f.jpg)
![6step](https://user-images.githubusercontent.com/67605008/86055791-9f43ce00-ba5c-11ea-8051-a95061c67bb9.jpg)
![final](https://user-images.githubusercontent.com/67605008/86055794-9fdc6480-ba5c-11ea-840b-f4a5fcfd9590.jpg)

**Comment by fmw42** [comment]
I am still not sure what you want. But here is how to add text using caption:

Input:

![img](https://user-images.githubusercontent.com/13458218/86062409-7c45f980-ba1d-11ea-8eb0-4b68f3451583.jpg)

Unix syntax:

```
convert img.jpg \
\( -size 760x -background none \
-fill "rgb(37,87,37)" -gravity center -pointsize 100 -font candice \
caption:"Anne of Green Gables" \) \
-geometry +1110+280 -gravity northwest -compose over -composite \
result.jpg

```

Result:

![result](https://user-images.githubusercontent.com/13458218/86062432-8bc54280-ba1d-11ea-8a07-f26fcee2ba82.jpg)

**Comment by jaka1-newbie** [comment]
How it centred in right part? Size of cover will be changed...

**Reply by fmw42** [reply]
You have to figure that out. There is nothing in ImageMagick that will automatically center to some arbitrary location. You simply specify an upper left corner and size of the area into which you want the text to be placed.

**Comment by jaka1-newbie** [comment]
Can we count from image width this? 

How fill with solid color part of image?

**Reply by fmw42** [reply]
Sorry, I do not understand. When you tell ImageMagick the upper left corner of a region, that is measured from the left edge to the right by X and from the top down along Y.

What do you want to fill with color? You can create a solid color image of any size and place that over the top of your image. Is that what you want?

**Comment by jaka1-newbie** [comment]
Yes. Solid color. 



Отправлено из мобильной Почты Mail.ru


вторник, 30 июня 2020 г., 01:41 +0200 от notifications@github.com  <notifications@github.com>:

**Comment by fmw42** [comment]
Here is an example. I have an image that I want to fill the top 25% with pink. So I read the image. Then make a copy and colorize it all with pink. Then I crop the top 25% and put that at the top of the original.

Input:

![logo](https://user-images.githubusercontent.com/13458218/86075429-d4402880-ba3c-11ea-9f31-5354820cb800.jpg)

```
convert logo.jpg \
\( +clone -fill pink -colorize 100 -crop 100x25%+0+0 \) \
-compose over -composite result.jpg
```

![result](https://user-images.githubusercontent.com/13458218/86075485-f46fe780-ba3c-11ea-8734-a36b79250429.jpg)

**Reply by jaka1-newbie** [reply]
This example didn't work with me.
I run like this:
 convert "D:\Dropbox\Z\My\test.jpg" +clone -fill pink -colorize 100 -crop 10x25%+0+0 -compose over -composite "D:\Dropbox\Z\My\result.jpg"

![result](https://user-images.githubusercontent.com/67605008/86111314-e666a900-bac6-11ea-895f-47385c75a602.jpg)
![test](https://user-images.githubusercontent.com/67605008/86111316-e6ff3f80-bac6-11ea-9ec4-3f9deb9eba78.jpg)

**Comment by jaka1-newbie** [comment]
Thx a lot! I start to understand.
Can we do with ImageMagick this effect or need to compose 2 images? 
![1](https://user-images.githubusercontent.com/67605008/86110151-7d326600-bac5-11ea-9108-7d4451ed260b.jpg)
![2](https://user-images.githubusercontent.com/67605008/86110154-7e639300-bac5-11ea-85ac-4a945faa194d.jpg)

**Comment by snibgo** [comment]
```
This example didn't work with me.
```
If you use Windows BAT scripts, you need to double the percent signs % like this:
```
convert "D:\Dropbox\Z\My\test.jpg" +clone -fill pink -colorize 100 -crop 10x25%%+0+0 -compose over -composite "D:\Dropbox\Z\My\result.jpg"
```

**Reply by jaka1-newbie** [reply]
I use from another programm. %% didn't work.
May be () anythere need to be?

**Comment by snibgo** [comment]
```
I use from another programm. 
```
What other program?

**Reply by jaka1-newbie** [reply]
Zennoposter

**Comment by snibgo** [comment]
Then you should consult Zennoposter documentation or a forum or whatever.

**Reply by jaka1-newbie** [reply]
The help says that the syntax is the same as on the command line

**Comment by snibgo** [comment]
You can add "-debug command" or "-debug all" to the IM command, and examine the text output to ensure IM is seeing the percent sign "%" etc.

**Reply by jaka1-newbie** [reply]
[Magick-0.log](https://github.com/ImageMagick/ImageMagick/files/4852575/Magick-0.log)

**Comment by jaka1-newbie** [comment]
Waiting for fmw42
If i put colorise 90 then i have this

Something wrong with compose
![result2](https://user-images.githubusercontent.com/67605008/86141988-f98f6e00-baf2-11ea-8afd-c9817b811965.jpg)

**Comment by snibgo** [comment]
From your log file
```
  Command line: convert {-debug} {all} {F:\Dropbox\Zenno\My\test.jpg} {+clone} {-fill} {pink} {-colorize} {100} {-crop} {100x25%+0+0} {-compose} {over} {-composite} {F:\Dropbox\Zenno\My\result2.jpg}</event>
```
The percentage sign is good.

I don't understand what you want. In your command, you make a copy of the input with "+clone". Then you colorize both images, so they remain the same. Then you crop both images, so they remain the same. Then you composite one over the other.

I don't suppose that is what you want.

**Reply by jaka1-newbie** [reply]
I can't repeat https://github.com/ImageMagick/ImageMagick/discussions/2212#discussioncomment-31753 
Code same result - NO

**Comment by snibgo** [comment]
The command fmw42 showed you had parentheses : ( and ). But you don't have them. Why not?

**Reply by jaka1-newbie** [reply]
Comand line doesn't support them...
Any advice?

**Comment by snibgo** [comment]
My advice is: you need them.

**Reply by jaka1-newbie** [reply]
Tried. Don't work

**Comment by jaka1-newbie** [comment]
parentheses : ( and ) are passed to the IM, but it processes them incorrectly.
So need to divide it into two stages.

**Comment by jaka1-newbie** [comment]
Can't place vertical text.
Please help.

**Reply by fmw42** [reply]
Create a horizontal text image using caption: or label:. Then rotate it -90. Then composite it over the center of the image using -gravity center -compose over -composite.

**Reply by jaka1-newbie** [reply]
Thx for your help! 
Any suggestion about parentheses : ( and )?

**Reply by fmw42** [reply]
You need to keep the parentheses. If on Windows, then do not need escaping. On Unix they need \ escapes. Using it it other tools is hard to tell you since I do not use that tool. But consider proper use of quotes or other escapes for your tool to make it work.
